xmtp_mls/src/groups/mod.rs

```diff
@@ -485,6 +484,7 @@ impl MlsGroup {
         if inbox_id_map.len() != account_addresses.len() {
             let found_addresses: HashSet<&String> = inbox_id_map.keys().collect();
             let to_add_hashset = HashSet::from_iter(account_addresses.iter());
+            self.pre_intent_hook(client).await?;
```

Comment:
@richardhuaaa I merged the new changes which use `inbox_id` instead and it seems we don't create new group intent in the `add_members` function - see [this commit ](https://github.com/xmtp/libxmtp/pull/658/commits/4a317b4d93b1013db101500f16d0d8eb42ac6474#diff-c29f56a38916c7410eff8091df1a2e43487ffe20646d96827e846e475f4608d3L480)for a more clear comparison of what changed.

The original code was applying the `pre_intent_hook` function before every new group intent created, but now since it seems there is no new group intent created anymore and we directly add members by inbox id, my question was that I'm not sure if we still need to apply `pre_intent_hook` to update the key anymore (?)

If we still need, where is the best place to apply the key update?

Reply:
So there must be an intent created somewhere, because intents are the only way that we can produce commits, and you can't add a member to a group without creating a commit.

It may have just been moved deeper down into the call tree, looks like it might be here? https://github.com/xmtp/libxmtp/blob/187173415812fc85977d6534df7d06d8b393918e/xmtp_mls/src/groups/mod.rs#L505

Maybe you can move your pre_intent_hook there? 

In general, it might make sense to search for all instances of `Intent::new` or `IntentData::new` to see all cases where we are creating intents.